clients/unshaded/src/main/java/tachyon/client/TachyonFile.java

```diff
@@ -497,7 +497,7 @@ boolean recache(int blockIndex) throws IOException {
       int bufferBytes =
           (int) mTachyonConf.getBytes(Constants.USER_FILE_BUFFER_BYTES, Constants.MB) * 4;
       byte[] buffer = new byte[bufferBytes];
-      bos = new BlockOutStream(this, WriteType.TRY_CACHE, blockIndex, mTachyonConf);
+      bos = new LocalBlockOutStream(this, WriteType.TRY_CACHE, blockIndex, mTachyonConf);
```

Comment:
Yes, this can be either a Local one or Remote one. I changed it so it will choose accordingly. I had originally thought this was only possible locally, but remote should be possible. Thanks!